spec/sec-regexp-regular-expression-objects-patch.html

```diff
@@ -280,71 +280,50 @@ <h1><ins>GetStringIndex ( _S_, _Input_, _e_ )</ins></h1>
-        1. If _fullUnicode_ is *true*, then
-          1. Assert: _matchChars_ is a List of code points.
-          1. Return the String value whose code units are the UTF16Encoding of the code points of _matchChars_.
-        1. Else _fullUnicode_ is *false*,
-          1. Assert: _matchChars_ is a List of code units.
-          1. Return the String value consisting of the code units of _matchChars_.
+        1. Return the portion of _S_ between offset _startIndex_ inclusive and offset _endIndex_ exclusive.
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-getmatchresult" aoid="GetMatchResult">
+    <emu-clause id="sec-coercematchresult" aoid="CoerceMatchResult">
       <h1><ins>CoerceMatchResult ( _match_, _matchOptions_ )</ins></h1>
       <p>The abstract operation CoerceMatchResult with arguments _match_ and _matchOptions_ performs the following steps:</p>
       <emu-alg>
       1. Assert: _matchOptions_ is a MatchOptions Record.
       1. If _matchOptions_.[[Capture]] is `"indices"`, then
-        1. Set _match_ to be ? ToObject(_match_).
-        1. Let _startIndex_ be ? Get(_match_, `"0"`).
-        1. Let _endIndex_ be ? Get(_match_, `"1"`).
-        1. If Type(_startIndex_) is not Number, throw a *TypeError* exception.
-        1. If Type(_endIndex_) is not Number, throw a *TypeError* exception.
-        1. Return CreateArrayFromList(&laquo; _startIndex_, _endIndex_ &raquo;).
-      1. Else,
-        1. Return ? ToString(_match_).
-      </emu-alg>
-    </emu-clause>
-
-    <emu-clause id="sec-isemptymatch" aoid="IsEmptyMatch">
-      <h1><ins>IsEmptyMatch ( _match_, _matchOptions_ )</ins></h1>
-      <p>The abstract operation IsEmptyMatch with arguments _match and _matchOptions_ performs the following steps:</p>
-      <emu-alg>
-      1. Assert: _matchOptions_ is a MatchOptions Record.
-      1. If _matchOptions_.[[Capture]] is `"indices"`, then
-        1. Set _match_ to be ? ToObject(_match_).
-        1. Let _startIndex_ be ? Get(_match_, `"0"`).
-        1. Let _endIndex_ be ? Get(_match_, `"1"`).
-        1. If Type(_startIndex_) is not Number, throw a *TypeError* exception.
-        1. If Type(_endIndex_) is not Number, throw a *TypeError* exception.
-        1. Return _endIndex_-_startIndex_=0.
+        1. If Type(_match_) is not Object, throw a *TypeError* exception.
+        1. Let _start_ be ? Get(_match_, `"0"`).
+        1. Let _end_ be ? Get(_match_, `"1"`).
+        1. If Type(_start_) is not Number, throw a *TypeError* exception.
+        1. If Type(_end_) is not Number, throw a *TypeError* exception.
+        1. Let _isEmpty_ be _start_ = _end_.
       1. Else,
-        1. Let _matchStr_ be ? ToString(_match_).
-        1. If _matchStr_ is the empty String, then
-          1. Return True.
-        1. Return False.
+        1. Set _match_ to ? ToString(_match_).
```

Comment:
Can spec algorithms mutate their parameters like this? The Algorithm Conventions section (5.2) only describes "Set _x_ to ..." applying to things defined by "Let _x_ be ..." earlier in the same algorithm, not to algorithm parameters.